jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/SizeLimitHandler.java

```diff
@@ -59,8 +59,11 @@ public boolean handle(Request request, Response response, Callback callback) thr
         if (contentLengthField != null)
         {
             long contentLength = contentLengthField.getLongValue();
-            if (contentLength > _requestLimit)
-                throw new BadMessageException(413, "Request body is too large: " + contentLength + ">" + _requestLimit);
+            if (_requestLimit >= 0 && contentLength > _requestLimit) {
+                String s = "Request body is too large: " + contentLength + ">" + _requestLimit;
+                Response.writeError(request, response, callback, 413, s);
+                return true;
+            }
```

Comment:
This might be better:
```suggestion
            if (_requestLimit >= 0 && contentLength > _requestLimit) {
                callback.fail(new BadMessageException("Request body is too large: " + contentLength + ">" + _requestLimit));
                return true;
            }
```